### ISSUE TITLE
Fatal error (3.1.30)

### ISSUE THREAD

**Issue by lhoumeau** [smarty-php/smarty#307]
With 3.1.30 only (not 3.1.29):

\nFatal error: unknown class Smarty_Internal_Block in /media/psf/Home/www/(nginx) components/php/prod/vendor/smarty/smarty/libs/sysplugins/smarty_template_compiled.php(133)(75bcc357ed31b34886eb831b45a594c9) : eval()'d code on line 31

**Comment by uwetews** [comment]
It looks like that you are using composer and updated from an older Smarty version not using Composer Update, but copied the distribution files.

In this case the composer autoloader is not updated to include the new class  Smarty_Internal_Block

**Comment by lhoumeau** [comment]
I don't remember the previous version before a simple "composer update" (~3.1). But I go back to 3.1.28 then 3.1.29 and it' fine for both of them.

**Comment by uwetews** [comment]
I tried to run "composer update" (~3.1)  on 3.1.28 and 3.1.29.

And everything worked

**Comment by lhoumeau** [comment]
Back to ~3.1 and the issue is here. But only on template with form. I'll check changelog with hope to find something about forms in 3.1.30 updates.

**Comment by lhoumeau** [comment]
My template pages for forms uses many smarty blocks or functions. My "input" smarty function for <input ...> seems to fail on : return $plug->fetch('extends:form.Control.tpl|file:control.input.tpl');
These templates are simple: only {$params.xxx} and some {if cond} {/if}, mixed with html tag.

**Comment by uwetews** [comment]
Coming back to the original error : unknown class Smarty_Internal_Block is caused  when the autoloader
can not load  the required PHP file.
1.)
The class is located  in smarty/lids/sysplugins/smarty_internal_block.php. Is the file present?
2.) 
After the upgrade with composer the array in vendor/composer/autoload_classmap.php must contain an entry for 'Smarty_Internal_Block'

Got you above error only on your "input" smarty function?
Which class if object $plug ?

Please try the dev master version with  "composer update" (~3.1.dev)
There is since today a change of fetch() calls on template object.
It could perhaps also solve your problem.

**Comment by lhoumeau** [comment]
1.) Yes, the file is present.
2.) No, there is no entry in autoload_classmap.php for 'Smarty_Internal_Block' (there is an entry for 'Smarty_Internal_Compile_Block').

Tried today with:
 ~3.1 (same message in error log)
 ~3.1.dev (new message: Fatal error: Uncaught  --> Smarty: undefined extension class 'Smarty_Internal_Runtime_CacheResourceFile' <--)

So, error pending.

Yes, the error appears only for pages that include my 'input" smarty function. And $plug = new Smarty_Plugin; (as all my function plugins).

Regards.

**Comment by uwetews** [comment]
Please crosscheck once again.
1, file is located at  /vendor/smarty/smarty/libs/sysplugins/smarty_internal_block.php ?
2. does contain class Smarty_Internal_Block ?

Could you provide the code of your 'input' smarty function and class Smarty_Plugin

**Comment by lhoumeau** [comment]
back to ~3.1 in composer:

1. smarty_internal_block.php is located /vendor/smarty/smarty/libs/sysplugins
2. class Smarty_Internal_Block included

Timestamp: strange. In /vendor/composer, only installed.json seems updated (correct timestamp). No other file timestamped according to installed.json timestamp. And I have found the cause : on 'composer update' process, I rejected another package update. In that case, /vendor/composer files are not updated. So I have run a full update of all packages ... and all /vendor/composer files are updated, so autoload_classmap.php now includes Smarty_Internal_Block entry and everything is OK (pages are displayed).

Solved for me. Thanks a lot!
